connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractWorkerSourceTask.java

```diff
@@ -397,7 +399,11 @@ boolean sendRecords() {
                     producerRecord,
                     (recordMetadata, e) -> {
                         if (e != null) {
-                            log.debug("{} failed to send record to {}: ", AbstractWorkerSourceTask.this, topic, e);
+                            if (producerClosed) {
```

Comment:
Added this check to clean up the flood of `ERROR`-level log messages when the task's producer is closed while there are still in-flight messages.